contracts/sidechain/L1PoolManagerProxy.sol

```diff
@@ -0,0 +1,123 @@
+
+    function vote_for_gauge_weights(address, uint256) external;
+
+    function add_gauge(
+        address,
+        int128,
+        uint256
+    ) external;
+
+    function gauges(uint256) external view returns (address);
+
+    function checkpoint_gauge(address) external;
+
+    function n_gauges() external view returns (uint256);
+}
+
+contract L1PoolManagerProxy is LzApp {
+    /// @dev LayerZero chain ID for this chain
+    uint16 public immutable lzChainId;
+    /// @dev Gauge controller address
+    address public immutable gaugeController;
+    /// @dev Indicates if add pool is protected or not.
+    bool public protectAddPool;
+
+    /* -------------------------------------------------------------------
+       Events 
+    ------------------------------------------------------------------- */
+    event AddSidechainPool(uint16 indexed dstChainId, address rootGauge, address dstGauge);
+
+    /* -------------------------------------------------------------------
+       Constructor 
+    ------------------------------------------------------------------- */
+
+    /**
+     * @param _lzChainId LayerZero chain ID
+     * @param _lzEndpoint LayerZero endpoint
+     * @param _gaugeController Gauge controller address
+     */
+    constructor(
+        uint16 _lzChainId,
+        address _lzEndpoint,
+        address _gaugeController
+    ) {
+        lzChainId = _lzChainId;
+        _initializeLzApp(_lzEndpoint);
+        protectAddPool = true;
+        gaugeController = _gaugeController;
+    }
+
+    receive() external payable {}
+
+    /**
+     * @notice set if addPool is only callable by operator
+     */
+    function setProtectPool(bool _protectAddPool) external onlyOwner {
+        protectAddPool = _protectAddPool;
+    }
+
+    /**
+     * @notice Send a message to add a pool on a sidechain.
+     * @param _gauge              The root gauge address.
+     * @param _dstChainId         The LayerZero destination chain ID eg optimism is 111
+     * @param _zroPaymentAddress  The LayerZero ZRO payment address
+     * @param _adapterParams      The adapter params
+     */
+    function addPool(
+        address _gauge,
+        uint16 _dstChainId,
+        address _zroPaymentAddress,
+        bytes memory _adapterParams
+    ) external payable returns (bool) {
+        if (protectAddPool) {
+            require(msg.sender == owner(), "!auth");
+        }
+
+        require(_dstChainId != lzChainId, "!dstChainId");
+        address dstGauge = IStakelessGauge(_gauge).getRecipient();
+        require(dstGauge != address(0), "!dstGauge");
+        //check that the pool as weight
+        uint256 weight = IGaugeController(gaugeController).get_gauge_weight(_gauge);
+        require(weight > 0, "must have weight");
+
+        _lzSend(
+            _dstChainId, ///////////// Destination chain (L2 chain)
+            abi.encode(dstGauge), ///////////////// Payload encode
+            payable(address(this)), // Refund address
```

Comment:
Refund address should probably be the sender, as this is possibly publicly callable the sender will loose their refund.

Reply:
thanks , i changed it